src/lib/utils/os_utils.cpp

```diff
@@ -614,6 +620,35 @@ void OS::page_named(void* page, size_t size) {
 #endif
 }
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void OS::set_thread_name(std::thread& thread, const char* const name) {
+   int r;
+   auto hdl = thread.native_handle();
+   #if defined(BOTAN_TARGET_OS_IS_LINUX)
+   r = pthread_setname_np(hdl, name);
+   #elif defined(BOTAN_TARGET_OS_IS_FREEBSD)
+   r = pthread_set_name_np(hdl, name);
```

Comment:
FreeBSD also seems to support `pthead_setname_np` (https://man.freebsd.org/cgi/man.cgi?query=pthread_set_name_np) which would reduce the number of conditionals.

Reply:
If we consider windows the only exception in term of supporting EOL releases then yes.